code/modules/antagonists/mind_flayer/powers/flayer_stealth_powers.dm

```diff
@@ -0,0 +1,256 @@
+	flayer.add_ability(new /datum/spell/flayer/self/voice_synthesizer)
+
+/datum/spell/flayer/skin_suit/on_apply()
+	..()
+	cooldown_handler.recharge_duration -= 30 SECONDS
+
+/datum/spell/flayer/skin_suit/Destroy(force, ...)
+	flayer?.remove_ability(/datum/spell/flayer/self/voice_synthesizer)
+	return ..()
+
+/// After a 7 second channel time you can emag a borg
+/datum/spell/flayer/self/override_key
+	name = "Silicon Administrative Access"
+	desc = "Allows us to charge our hand with a mass of nanites that hijacks cyborgs lawsets."
+	action_icon_state = "magnet" // Uhhhhhhhhhhhhhhhhhhhhhhhhhhh
+	power_type = FLAYER_PURCHASABLE_POWER
+	category = FLAYER_CATEGORY_INTRUDER
+	base_cooldown = 2 SECONDS //The cast time is going to be the main limiting factor, not cooldown
+	base_cost = 150
+	stage = 3
+	var/hand_type = /obj/item/melee/swarm_hand
+
+/datum/spell/flayer/self/override_key/cast(list/targets, mob/user)
+	if(istype(user.l_hand, hand_type))
+		qdel(user.l_hand)
+		flayer.send_swarm_message("We dissipate the nanites.")
+		return
+	if(istype(user.r_hand, hand_type))
+		qdel(user.r_hand)
+		flayer.send_swarm_message("We dissipate the nanites.")
+		return
+
+	var/obj/item/melee/swarm_hand/funny_hand = new hand_type
+	if(!user.put_in_hands(funny_hand))
+		flayer.send_swarm_message("Our hands are currently full.")
+		qdel(funny_hand)
+		return
+
+/obj/item/melee/swarm_hand
+	name = "Nanite Mass"
+	desc = "Will attempt to convert any cyborg you touch into a loyal member of the hive after a 7 second delay."
+	icon = 'icons/obj/weapons/magical_weapons.dmi'
+	lefthand_file = 'icons/mob/inhands/items_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items_righthand.dmi'
+	icon_state = "disintegrate"
+	item_state = "disintegrate"
+	color = COLOR_BLACK
+	flags = ABSTRACT | DROPDEL
+	w_class = WEIGHT_CLASS_HUGE
+	var/conversion_time = 7 SECONDS
+
+/obj/item/melee/swarm_hand/afterattack(atom/target, mob/living/user, proximity_flag, click_parameters)
+	. = ..()
+	if(!isrobot(target))
+		return
+	var/mob/living/silicon/robot/borg = target
+	target.visible_message(
+		"<span class='danger'>[user] puts [user.p_their()] hands on [target] and begins transferring energy!</span>",
+		"<span class='userdanger'>[user] puts [user.p_their()] hands on [target] and begins transferring energy!</span>")
+	if(borg.emagged || !borg.is_emaggable)
+		to_chat(user, "<span class='notice'>Your override attempt fails before it can even begin.</span>")
+		qdel(src)
+		return
+	if(!do_mob(user, borg, conversion_time))
+		to_chat(user, "<span class='notice'>Your concentration breaks.</span>")
+		qdel(src)
+		return
+	to_chat(user, "<span class='notice'>The mass of swarms vanish into the cyborg's internals. Success.</span>")
+	INVOKE_ASYNC(src, PROC_REF(emag_borg), borg, user)
+	qdel(src)
+
+/obj/item/melee/swarm_hand/proc/emag_borg(mob/living/silicon/robot/borg, mob/living/user)
+	if(QDELETED(borg) || QDELETED(user))
+		return
+	sleep(6)
```

Comment:
Why the sleep here at the start? Why not just replace the INVOKE_ASYNC with a timer?